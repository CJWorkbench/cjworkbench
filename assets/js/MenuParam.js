// Simple wrapper over HTML <select>
import React from 'react'
import PropTypes from 'prop-types'

export default class MenuParam extends React.Component {
  constructor(props) {
    super(props);
    this.onChange = this.onChange.bind(this);
    this.state = {
      selectedIdx: this.props.selectedIdx
    };
  }
/*
  componentWillReceiveProps(newProps) {
    if (this.state.selectedIdx != newProps.selectedIdx)
      this.setState({selectedIdx :  newProps.selectedIdx})
  }
*/
  onChange(evt) {
    var idx =  evt.target.value;
    this.setState({selectedIdx: idx });
    this.props.onChange(idx);
  }

  render() {
    var items = this.props.items.split('|');
    var itemDivs = items.map( (name, idx) => {
        return <option key={idx} value={idx}>{name}</option>;
    });

    return (
      <div>
        <label className='mr-1'>{this.props.name}:</label>
        <select className="custom-select" value={this.state.selectedIdx} onChange={this.onChange}>
          {itemDivs}
        </select>
      </div>
    );
  }
}

MenuParam.propTypes = {
  name:         PropTypes.string,
  items:        PropTypes.string,  // like 'Apple|Banana|Kitten'
  selectedIdx:  PropTypes.number,
  onChange:     PropTypes.func     // called with index of selected item
};
