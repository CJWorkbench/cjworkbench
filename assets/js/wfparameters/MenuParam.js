// Simple wrapper over HTML <select>
import React from 'react'
import PropTypes from 'prop-types'

export default class MenuParam extends React.Component {
  constructor(props) {
    super(props);
    this.onChange = this.onChange.bind(this);
    this.state = {
      selectedIdx: this.props.selectedIdx
    };
  }
/*
  componentWillReceiveProps(newProps) {
    if (this.state.selectedIdx != newProps.selectedIdx)
      this.setState({selectedIdx :  newProps.selectedIdx})
  }
*/
  onChange(evt) {
    var idx =  evt.target.value;
    this.setState({selectedIdx: idx });
    this.props.onChange(idx);
  }

  render() {
    var items = this.props.items.split('|');
    var itemDivs = items.map( (name, idx) => {
        return <option key={idx} value={idx} className='dropdown-menu-item t-d-gray content-3'>{name}</option>;
    });

    return (
      <div className='parameter-margin d-flex flex-column'>
        <label className='label-margin t-d-gray content-3'>{this.props.name}</label>
        <select
          className='custom-select dropdown-selector'
          value={this.state.selectedIdx}
          onChange={this.onChange}
          disabled={this.props.isReadOnly}
        >
          {itemDivs}
        </select>
      </div>
    );
  }
}

MenuParam.propTypes = {
  name:         PropTypes.string,
  items:        PropTypes.string,  // like 'Apple|Banana|Kitten'
  selectedIdx:  PropTypes.number,
  onChange:     PropTypes.func     // called with index of selected item
};
