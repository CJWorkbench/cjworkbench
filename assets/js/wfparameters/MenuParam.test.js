import React from 'react'
import { shallow, render, mount } from 'enzyme'

import MenuParam from './MenuParam';

it('renders correctly', () => {
  var menuStr = 'Apple|Kittens|Banana';

  const wrapper = shallow( <MenuParam items={menuStr} name="SuperMenu" selectedIdx={1} onChange={ () => {} } /> );
  expect(wrapper).toMatchSnapshot();
});

