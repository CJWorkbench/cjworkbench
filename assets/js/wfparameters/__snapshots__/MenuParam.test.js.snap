// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<select
  className="custom-select module-parameter dropdown-selector"
  name="SuperMenu"
  onChange={[Function]}
  value={1}
>
  <option
    className="dropdown-menu-item t-d-gray content-3"
    key="0"
    value={0}
  >
    Apple
  </option>
  <option
    className="dropdown-menu-item t-d-gray content-3"
    key="1"
    value={1}
  >
    Kittens
  </option>
  <option
    className="dropdown-menu-item t-d-gray content-3"
    key="2"
    value={2}
  >
    Banana
  </option>
</select>
`;
